File: src/main/java/com/dh/ondot/member/domain/repository/AddressRepository.java
package com.dh.ondot.member.domain.repository;

import com.dh.ondot.member.domain.Address;
import com.dh.ondot.member.domain.enums.AddressType;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface AddressRepository extends JpaRepository<Address, Long> {
    Optional<Address> findByMemberIdAndType(Long memberId, AddressType type);
    void deleteByMemberId(Long memberId);
}

File: src/main/kotlin/com/dh/ondot/member/domain/repository/AddressRepository.kt
package com.dh.ondot.member.domain.repository

import com.dh.ondot.member.domain.Address
import com.dh.ondot.member.domain.enums.AddressType
import org.springframework.data.jpa.repository.JpaRepository

interface AddressRepository : JpaRepository<Address, Long> {
    fun findByMemberIdAndType(memberId: Long, type: AddressType): Address?
    fun deleteByMemberId(memberId: Long)
}
